src/contexts/SolanaWalletContext.tsx

```diff
@@ -20,13 +21,40 @@ import {
   BloctoWalletAdapter,
   BraveWalletAdapter,
 } from "@solana/wallet-adapter-wallets";
-import { FC, useMemo } from "react";
 import { CLUSTER, SOLANA_HOST } from "../utils/consts";
 
+declare global {
+  interface Navigator {
+    brave: {
+      isBrave(): Promise<boolean>;
+    };
+  }
+  interface Global {
+    navigator: Navigator;
+  }
+}
+
+export const isBraveBrowser = async () => {
+  if (window.navigator.brave) {
+    return await window.navigator.brave.isBrave();
+  }
+  return false;
+};
+
 export const SolanaWalletProvider: FC = (props) => {
+  const [isBrave, setIsBrave] = useState<boolean | null>(null);
+
+  useEffect(() => {
+    const checkIfIsBraveBrowser = async () => {
+      const getIsBrave: boolean = await isBraveBrowser();
+      setIsBrave(getIsBrave);
+    };
+
+    checkIfIsBraveBrowser();
+  }, [isBrave]);
```

Comment:
I think this useEffect should be fired only once on mount, so the dependency array should be empty
(also it's checking on isBrave and this effect sets the isBrave state)

Reply:
fixed